src/matcher/compound.js

```diff
@@ -1,19 +1,17 @@
 'use strict';
 
-function Matcher (matchers) {
-  this._matchers = matchers;
-}
+module.exports = function ($fs, matchers) {
+  matchers = matchers.map(function (matcher) {
+    return matcher($fs);
+  });
 
-Matcher.prototype = {
-  match: function (code) {
+  return function (file, code) {
     var matches = [];
 
-    this._matchers.forEach(function (matcher) {
-      matches = matches.concat(matcher.match(code));
+    matchers.forEach(function (matcher) {
```

Comment:
This can be written as a reduce (of `concat` with `[]` as the initial value).

I definitely prefer using reduce, map, etc., where appropriate since when I see `forEach()` I just about assume that there are side-effects about to happen.